pkgs/top-level/all-packages.nix

```diff
@@ -4691,7 +4691,7 @@ let
   sdcc = callPackage ../development/compilers/sdcc { };
 
   smlnjBootstrap = callPackage ../development/compilers/smlnj/bootstrap.nix { };
-  smlnj = callPackage_i686 ../development/compilers/smlnj { };
+  smlnj = callPackage ../development/compilers/smlnj { };
```

Comment:
That sounds plausible. Could we replicate that aspect of `callPackage_i686` in the `smlnj` derivation itself?